cli/src/main/java/hudson/cli/CLI.java

```diff
@@ -563,6 +568,17 @@ public boolean verify(String s, SSLSession sslSession) {
             return -1;
         }
 
+        if (auth == null) {
+            // -auth option not set
+            if (StringUtils.isNotBlank(userIdEnv) && StringUtils.isNotBlank(tokenEnv)) {
+                auth = StringUtils.defaultString(userIdEnv).concat(":").concat(StringUtils.defaultString(tokenEnv));
+            } else if (StringUtils.isBlank(userIdEnv) || StringUtils.isBlank(tokenEnv)) {
```

Comment:
I think this is not right now? If both are blank, the else if would apply and I don't think that's what you want.

Reply:
Before correcting, please add a test :)

Reply:
@varyvol is right and I shouldn't change the check (Sorry, I did without thinking). Just see my next comment `// Otherwise, none credentials were set`
@Wadeck the test exists (See the `// Without -auth option and without env vars` test) but as I'm using the `who-am-i` command, this error didn't appear. I decided use that CLI command because it allows us to know if the authentication is correct.

Reply:
> // Without -auth option and without env vars

I meant a test with only one of the two variables, that would have detected the bug :)